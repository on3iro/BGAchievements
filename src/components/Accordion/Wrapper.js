import styled from 'styled-components'

const Wrapper = styled.div`
  width: 100%;
  height: 90%;

  overflow: auto;

  div:last-child {
    margin-bottom: 0;
  }
`

export default Wrapper
