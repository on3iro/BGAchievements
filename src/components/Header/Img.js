/**
  * Renders a styled img
  *
  * @namespace Header.Img
  * @memberOf Header
  */

import styled from 'styled-components'

import NormalImg from 'components/Img'

const Img = styled(NormalImg)`
  width: 70px;
  display: block;
`

export default Img
