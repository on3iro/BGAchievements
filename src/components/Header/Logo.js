/**
  * Renders a reformatted anchor
  *
  * @namespace Header.A
  * @memberOf Header
  */

import styled from 'styled-components'
import { Link } from 'react-router-dom'

const Logo = styled(Link)`
  display: inline-block;
  margin: 10px 80px;
  outline: 0;
`

export default Logo
