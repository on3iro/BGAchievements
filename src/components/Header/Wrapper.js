/**
  * Styled container for Header
  *
  *
  * @namespace Header.Wrapper
  * @memberOf Header
  */

import styled from 'styled-components'

const Wrapper = styled.header`
  position: fixed;
  top: 0px;
  margin-bottom: 0;

  min-height: 64px;
  height: 64px;
  width: 100%;

  background-color: ${props => props.theme.mainBG};
  color: ${props => props.theme.noneWhite};

  @media (max-width: 900px) {
    position: static;

    height: auto;
  }
`

export default Wrapper
