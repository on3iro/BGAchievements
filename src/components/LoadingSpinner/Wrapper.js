import styled from 'styled-components'

const Wrapper = styled.div`
  width: 100%;
  height: 100%;
  text-align: center;
`

export default Wrapper
