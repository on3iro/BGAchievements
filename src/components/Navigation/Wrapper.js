import styled from 'styled-components'

export default styled.nav`
  margin: 12px 50px 15px 0;
  height: 100%;

  text-align: right;
`
