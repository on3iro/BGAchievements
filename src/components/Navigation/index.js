/**
  * Renders a Navigation with react-router Links
  *
  * @namespace Navigation
  */

import React from 'react'
import PropTypes from 'prop-types'

import Wrapper from './Wrapper'
import NavBar from './NavBar'
import NavLink from 'components/NavLink'

const Navigation = props => {
  return (
    <Wrapper>
      <NavBar>
        {
          props.isLoggedIn &&
            <NavLink to='/home'>
              Home
            </NavLink>
        }
      </NavBar>
    </Wrapper>
  )
}

Navigation.propTypes = {
  isLoggedIn: PropTypes.bool
}

export default Navigation
