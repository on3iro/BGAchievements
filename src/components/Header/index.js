/**
  * Renders a Header with a Logo and a navigation
  *
  * @namespace Header
  */

import React from 'react'
import Grid from 'grid-styled'

import Navigation from 'components/Navigation'
import UserNav from 'components/UserNav'

import Logo from './Logo'
import Img from './Img'
import Wrapper from './Wrapper'

export const Header = props => {
  return (
    <Wrapper>
      <Grid sm={1 / 1} md={1 / 6}>
        <Logo to='/'>
          <Img src='' alt='placeholder' />
        </Logo>
      </Grid>
      <Grid sm={1 / 1} md={4 / 6}>
        <Navigation {...props} />
      </Grid>
      <Grid sm={1 / 1} md={1 / 6}>
        <UserNav {...props} />
      </Grid>
    </Wrapper>
  )
}

export default Header
