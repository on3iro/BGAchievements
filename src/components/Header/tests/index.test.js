import React from 'react'
import { shallow } from 'enzyme'

import Header from '../index'
import Logo from '../Logo'
import Wrapper from '../Wrapper'
import Navigation from 'components/Navigation'

describe('<Header />', () => {
  it('should render a Wrapper', () => {
    const renderedComponent = shallow(<Header />)
    expect(renderedComponent.find(Wrapper).length).toMatchSnapshot()
  })

  it('Should render an <Logo />', () => {
    const wrapper = shallow(<Header />)
    expect(wrapper.find(Logo).length).toMatchSnapshot()
  })

  it('Should render a <Navigation />', () => {
    const wrapper = shallow(<Header />)
    expect(wrapper.find(Navigation).length).toMatchSnapshot()
  })
})
