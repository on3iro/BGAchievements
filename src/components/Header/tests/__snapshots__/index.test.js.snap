exports[`<Header /> Should render a <Navigation /> 1`] = `1`

exports[`<Header /> Should render an <Logo /> 1`] = `1`

exports[`<Header /> should render a Wrapper 1`] = `1`
